### PR TITLE
improve mechanics and test coverage for largeArraySize-option

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/array.test.js [hunk 1]
```diff
@@ -530,3 +530,86 @@ buildTest({
   largeArraySize: 2e4,
   largeArrayMechanism: 'default'
 })
+
+test('error on invalid value for largeArraySize /1', (t) => {
```
Comment (climba03003) [inline test/array.test.js:534]: I would use a `for-loop` to generate test-case.
Split in multiple one with same structure is not good for expanding when we need.

Comment (Uzlopak) [reply test/array.test.js:534]: I would like to keep it simple. I am just trying to improve test coverage.

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
Are the benchmarks impacted?

**Comment by Uzlopak** [comment]
I dont think so. Maybe the startup time insignificantly.

**Review by ivan-tymoshenko — APPROVED** [review]
lgtm

**Pull request merged**